pkg/services/ngalert/models/receivers.go

```diff
@@ -148,9 +147,39 @@ type IntegrationConfig struct {
 // IntegrationField represents a field in an integration configuration.
 type IntegrationField struct {
 	Name   string
+	Fields map[string]IntegrationField
 	Secure bool
 }
 
+type IntegrationFieldPath []string
+
+func NewIntegrationFieldPath(path string) IntegrationFieldPath {
```

Comment:
This simplified the code for traversing nested objects and avoided passing strings or slices around.